internal/pulsarutils/async.go

```diff
@@ -11,10 +11,25 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/G-Research/armada/internal/common/logging"
-	"github.com/G-Research/armada/internal/lookoutingester/model"
-	"github.com/G-Research/armada/pkg/armadaevents"
 )
 
+// ConsumerMessageId wraps a pulsar message id  and an identifier for the consumer which originally received the
+// corresponding message.  This exists because we need to track which messages came from which consumers so that
+// we can ACK them on the correct consumer.
+type ConsumerMessageId struct {
```

Comment:
So we can have concurrency from multiple pulsar readers inside a single process.  I'm not convinced we absolutely need it and I currently default this concurrency to 1, but it wasn't too hard to add and I think it may come in useful one day.